app/src/main/java/kth/jjve/xfran/EventPlanActivity.java

```diff
@@ -107,6 +110,33 @@ protected void onCreate(Bundle savedInstanceState) {
             datePickerDialog.show();
         });
 
+        /*------ TIME PICKER DIALOG -----*/
+        // TODO: see why you need to press the edittext twice
+        // TODO: see if can be implemented in method?
+        eventStartTimeEdit.setOnClickListener(v -> {
+            Calendar mCurrentTime = Calendar.getInstance();
+            int hour = mCurrentTime.get(Calendar.HOUR_OF_DAY);
+            int minute = mCurrentTime.get(Calendar.MINUTE);
+            TimePickerDialog timePickerDialog;
+            timePickerDialog = new TimePickerDialog(EventPlanActivity.this, (view, hourOfDay, minuteOfHour) -> {
+                @SuppressLint("DefaultLocale") String s_time = String.format("%02d:%02d", hourOfDay, minuteOfHour);
+                eventStartTimeEdit.setText(s_time);
+            }, hour, minute, true);//Yes 24 hour time
+            timePickerDialog.show();
+        });
```

Comment:
For some reason it doesn't work on my phone. Did you try yours or just the emulator?

Reply:
Works on both...